aspnet/fundamentals/logging/sample/src/LoggingSample/RequestLoggerMiddleware.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Logging.Internal;
+
+namespace LoggingSample
+{
+    public class RequestLoggerMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private readonly static Random _random = new Random();
+
+        public RequestLoggerMiddleware(RequestDelegate next, 
+            ILoggerFactory loggerFactory)
+        {
+            _next = next;
+            _logger = loggerFactory.CreateLogger<RequestLoggerMiddleware>();
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var eventId = _random.Next();
```

Comment:
event IDs are static. In general each actual log statement in code should have its own unique id, which doesn't vary.